### PR TITLE
Doc comments

### PR DESCRIPTION
**Pull request by mbr**
Adds doc comments parsing and output for both elm and Rust.

### CODE REVIEW

**Comment by problame** [comment]
Pushed a fix to this branch that makes the `chrono` timezone types work again.
With that patch applied, I verified that the generated rust code compiles.
Did not test the elm code.
LGTM

**Comment by problame** [comment]
Actually, I withdraw that: non-existent doc comments should not be rendered:

```
pub enum MonsterError {
    #[doc = ""]
    TooWeak,
    #[doc = ""]
    TooStrong {
        #[doc = ""]
        max_strength: i32,
    },
}
```

File: src/rust.rs [hunk 1]
```diff
@@ -24,10 +29,12 @@ pub fn render(spec: &ast::Spec) -> TokenStream {
 /// Render a struct definition.
 fn render_struct_def(sdef: &ast::StructDef) -> TokenStream {
     let ident = fmt_ident(&sdef.name);
+    let doc_comment = fmt_opt_string(&sdef.doc_comment);
     let fields: Vec<_> = sdef.fields.iter().map(render_pub_field_node).collect();
 
     quote!(
         #[derive(Debug, Clone, serde::Deserialize, serde::Serialize)]
+        #[doc = #doc_comment]
```
Comment (problame) [inline src/rust.rs:37]: this should only be rendered if `!doc_comment.is_empty()`.

Comment (mbr) [reply src/rust.rs:37]: I don't think this is a blocking issue for the generated code. Turning this into an issue.

**Pull request merged**